pkg/providers/imagefamily/bootstrap/aksbootstrap.go

```diff
@@ -95,138 +96,144 @@ func (a AKS) Script() (string, error) {
+	CustomSearchDomainFilepath        string            // s   static
+	HTTPProxyURLs                     string            // c   user input [presumably cluster-level]
+	HTTPSProxyURLs                    string            // c   user input [presumably cluster-level]
+	NoProxyURLs                       string            // c   user input [presumably cluster-level]
+	TLSBootstrappingEnabled           bool              // s   static true
+	SecureTLSBootstrappingEnabled     bool              // s   static false
+	DHCPv6ServiceFilepath             string            // k   derived from user input [how?]
+	DHCPv6ConfigFilepath              string            // k   derived from user input [how?]
+	THPEnabled                        string            // c   user input [presumably cluster-level][should be bool?]
+	THPDefrag                         string            // c   user input [presumably cluster-level][should be bool?]
+	ServicePrincipalFileContent       string            // s   only required for RP cluster [static: msi?]
+	KubeletClientContent              string            // -   unnecessary [if using TLS bootstrapping]
+	KubeletClientCertContent          string            // -   unnecessary
+	KubeletConfigFileEnabled          bool              // s   can be static	[should kubelet config be actually used/preferred instead of flags?]
+	KubeletConfigFileContent          string            // s   mix of user/static/RP-generated.
+	SwapFileSizeMB                    int               // t   user input
+	GPUImageSHA                       string            // s	  static sha rarely updated
+	GPUDriverVersion                  string            // k   determine by OS + GPU hardware requirements; can be determined automatically, but hard. suggest using GPU operator.
+	GPUInstanceProfile                string            // t   user-specified
+	CustomSearchDomainName            string            // c   user-specified [presumably cluster-level]
+	CustomSearchRealmUser             string            // c   user-specified [presumably cluster-level]
+	CustomSearchRealmPassword         string            // c   user-specified [presumably cluster-level]
+	MessageOfTheDay                   string            // t   user-specified [presumably node-level]
+	HasKubeletDiskType                bool              // t   user-specified [presumably node-level]
+	NeedsCgroupV2                     bool              // k   can be automatically determined
+	SysctlContent                     string            // t   user-specified
+	TLSBootstrapToken                 string            // X   nodepool or node specific. can be created automatically
+	KubeletFlags                      string            // psX unique per nodepool. partially user-specified, static, and RP-generated
+	KubeletFlagsMap                   map[string]string // Devin created for temp use. psX unique per nodepool. partially user-specified, static, and RP-generated
```

Comment:
Can you explain why this is needed some more? not sure I understand

Reply:
Oh this is because KubeletFlags in our contract is a KubeletFlagsMap while nbv.KubeletFlags is a string which consists of multiple key-value pairs indeed. That's why I created KubeletFlagsMap for now.
In this milestone, we likely will change it to a Map. (In the new contract it's a map now). So probably will remove the other one KubeletFlags.